tfhe/src/core_crypto/algorithms/lwe_encryption.rs

```diff
@@ -933,7 +1063,7 @@ pub fn encrypt_lwe_ciphertext_with_seeded_public_key<Scalar, KeyCont, OutputCont
         let (mut mask, body) = tmp_zero_encryption.get_mut_mask_and_body();
         random_generator
             .fill_slice_with_random_uniform_custom_mod(mask.as_mut(), ciphertext_modulus);
-        if !ciphertext_modulus.is_native_modulus() {
+        if ciphertext_modulus.is_non_native_power_of_two() {
             slice_wrapping_scalar_mul_assign(
                 mask.as_mut(),
                 ciphertext_modulus.get_power_of_two_scaling_to_native_torus(),
```

Comment:
I think it would be cleaner if  `fill_slice_with_random_uniform_custom_mod` returned a scaled mask

Reply:
If it's already the case, the multiplication should be removed here

Reply:
no the random should behave properly with respect to the modulus as we expose those functions IIRC, we have a special handling of the power of 2 at encryption and decryption time and that's on us to manage those cases, the rest should behave properly without our special encoding